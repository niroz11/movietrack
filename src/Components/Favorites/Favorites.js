import React from "react";
import { Component } from "react";
import { connect } from "react-redux";
import { Card } from '../Card/Card';
import { updateUser } from '../../actions/actions'


export class Favorites extends Component {
    constructor(props){
        super(props)}
    render(){
        console.log(this.props.user, "fav user")
        let displayFavorites;
        if(this.props.favoriteMovies && this.props.favoriteMovies.length > 0){
            displayFavorites = this.props.favoriteMovies.map((eachMovie) => {
                return <Card key={eachMovie.id} 
                             movies={eachMovie} 
                             id={eachMovie.movie_id} 
                             user={this.props.user}
                             updateUser={this.props.updateUser}/>
            })
        } else {
            return <p>No favs to display</p>
        }
      return (
          <div>
              {displayFavorites}
          </div>
      )}
}

export const mapStateToProps = (state) => ({
    favoriteMovies: state.user.favorites,
    user: state.user.id,
    email: state.user.email,
    user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
    updateUser: (id, name, favorites) => dispatch(updateUser(id, name, favorites))
})

export default connect(mapStateToProps, mapDispatchToProps)(Favorites)