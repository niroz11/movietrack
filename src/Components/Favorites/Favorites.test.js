import React from "react";
import { Favorites, mapStateToProps, mapDispatchToProps } from '../Favorites/Favorites'
import { shallow } from 'enzyme';


describe("Favorites", () => {
  describe("Favorites", () => {
    let wrapper;
  const mockUserFavorites = [
    { title: "Cars", poster_path: "car.jpg" },
    { title: "How To Train Your Dragon", poster_path: "dragon.jpg" }
  ];
  const mockUpdateUser = jest.fn();

  const mockUser = '1';
  const mockEmail = "abc@xyz.com"
  // const mockUser = 'tman'
  const mockId = '2'
  beforeEach(() => {
    wrapper = shallow(<Favorites favoriteMovies={mockUserFavorites}
                                 user={mockUser}
                                 email={mockEmail}
                                 updateUser={mockUpdateUser}
                                 id={mockId}/>)
  })

  it ('should match snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
  })

  describe("mapStateToProps", () => {
    
    it('should return users favorite movies', () => {
      const mockState = {
        favoriteMovies: [{"name": "django"}, {"year": "2019"}],
        user: "1",
        email: "abc@xyz.com",
        user: "taylor"
      }

      const expected = {
        favoriteMovies: [{"name": "django"}, {"year": "2019"}],
        user: "1",
        email: "abc@xyz.com",
        user: "taylor"
        }
        const mappedProps = mapStateToProps(mockState)
        expect(mappedProps).toEqual(expected)
        
        
   

    })
  })
  
})
