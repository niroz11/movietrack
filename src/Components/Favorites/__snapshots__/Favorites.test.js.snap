// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Favorites should match snapshot with all the data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    email="abc@xyz.com"
    favoriteMovies={
      Array [
        Object {
          "poster_path": "car.jpg",
          "title": "Cars",
        },
        Object {
          "poster_path": "dragon.jpg",
          "title": "How To Train Your Dragon",
        },
      ]
    }
    id="2"
    updateUser={[MockFunction]}
    user="1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          movies={
            Object {
              "poster_path": "car.jpg",
              "title": "Cars",
            }
          }
          user="1"
        />,
        <Card
          movies={
            Object {
              "poster_path": "dragon.jpg",
              "title": "How To Train Your Dragon",
            }
          }
          user="1"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": undefined,
          "movies": Object {
            "poster_path": "car.jpg",
            "title": "Cars",
          },
          "updateUser": undefined,
          "user": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": undefined,
          "movies": Object {
            "poster_path": "dragon.jpg",
            "title": "How To Train Your Dragon",
          },
          "updateUser": undefined,
          "user": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            movies={
              Object {
                "poster_path": "car.jpg",
                "title": "Cars",
              }
            }
            user="1"
          />,
          <Card
            movies={
              Object {
                "poster_path": "dragon.jpg",
                "title": "How To Train Your Dragon",
              }
            }
            user="1"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": undefined,
            "movies": Object {
              "poster_path": "car.jpg",
              "title": "Cars",
            },
            "updateUser": undefined,
            "user": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": undefined,
            "movies": Object {
              "poster_path": "dragon.jpg",
              "title": "How To Train Your Dragon",
            },
            "updateUser": undefined,
            "user": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
